### ISSUE TITLE
Improvement for fast DIN access

### ISSUE THREAD

**Issue by synatree** [pi-plates/ppDISTRO#1]
I have made some local changes to your driver to work around an issue I was facing with getDINbit, where sometimes it would crash the process.  Should I fork and submit a PR?

Alternatively, maybe I can describe the issue and let you sort out a better solution?  Mine is a hacky work-around.

**Comment by pi-plates** [comment]
Yes I'd like to hear about it since getDIN is a pretty benign operation.

**Comment by synatree** [comment]
It seems that sometimes, when accessing the DINs in a rapid manner (not that rapid really, but I guess I only noticed the problem when I had gotten up to a decent level of complexity), sometimes I would get back a [] (empty array) from a getDIN request.  It did not throw an error that was bubbling up to calling code, and as a result I ended up having to add a little logic to my code, in these cases, to just keep retrying until I got a zero or one.  

I don't know what would cause that, but it seems to be related to having more than one program using the same PiPlate at the same time (yes, I was careful to lock access exclusively between them).  For the most part it works, except when I see this behavior.

It's random, but not infrequent.
